### PR TITLE
Purchases: Update the list view design

### PR DESCRIPTION
**Pull request by ryelle**
See #12746 — This PR implements the new purchases list design. There should be no functionality change in this PR, though some bigger changes were made for passing lint (added localize, and renamed folders to pass the CSS classname check).

**Current**

screen shot 2017-04-05 at 1 48 01 pm

**Updated styles**

screen shot 2017-04-05 at 3 11 27 pm

Domain mapping/domain products use the domains gridicon…

screen shot 2017-04-05 at 3 14 31 pm

…while legacy products don't show an icon.

**To test**

1. Visit [/me/purchases](https://calypso.live/me/purchases?branch=update/purchases-list-design-i2)
2. Make sure it looks good 😄

### CODE REVIEW

**Comment by rickybanister** [comment]
I tweaked the styles a bit, making the plan icon half the size (28px) and adjusted the typography a little bit.

I also noticed we were using the 'spam' icon instead of the 'notice' icon which is very similar, but slightly different. We want 'notice' for telling users that their purchase is expiring, not the spam icon which looks more like a stopsign. 

image

image

**Comment by ryelle** [comment]
I've (hopefully?) fixed the issue that circleCI was complaining about, so I think this is ready to properly review — now that the site icon was also added.

**Comment by rickybanister** [comment]
I can help reduce the size of the site icon tomorrow—I think it needs to shrink a little to fit within the standard height of the section header. 

Looking good!

**Comment by rickybanister** [comment]
I wrangled the styles a bit now that we're using the `Site` instead of just sticking the icon into SectionHeader.

image

- [ ] we still need to update the placeholder, but I couldn't figure that out.

**Comment by MichaelArestad** [comment]
This looks much better. Are you going to mess with the spacing here?

image

Specifically, the space from the left side of the icon to the side of the card is much larger than the space from the top of the icon to the top of the card.

**Comment by rickybanister** [comment]
That's the standard spacing for a compact card

**Comment by rickybanister** [comment]
The placeholder looks great. I think we can merge this.

The only changes I'd suggest are updating the plan names to include 'Jetpack' in the title for Jetpack sites, but I gather that's not part of this codebase..

**Comment by ryelle** [comment]
I've added one more thing, because there could be "orphan" purchases on disconnected sites, and `Site` won't render for those. So that looks like this right now:

screen shot 2017-04-07 at 2 35 39 pm

I would suggest merging this as-is, and then doing an iteration to make it more obvious that it's a disconnected site that they're still paying for.

**Comment by rickybanister** [comment]
I concur—we can add a compact notice to the purchase saying `This site has been disconnected from WordPress.com` or something. Just like the expiration notice. It could perhaps have a 'reconnect' notice action that would attempt to direct them to the remote site to activate Jetpack (jump into the middle of JPC basically). 

Shall I make an issue for that?

**Comment by rickybanister** [comment]
Actually, a non-compact notice might be best since 'disconnection' relates to the site, not the individual purchase. We would want to attach the notice beneath the entire site block.

**Comment by ryelle** [comment]
Yeah, can you make an issue, and link it to #12746 ? That sounds good 🙂

**Comment by rickybanister** [comment]
Issue created: #12923

**Comment by tyxla** [comment]
@ryelle this is great stuff! I'd love to review that one, but there is a bunch of code that is moved from one file to another, and that makes difficult to track just the changes. Would it be possible to separate the file/directory renaming from the rest of the changes in two different PRs? Otherwise we could easily miss something. I'm fine with reviewing it as-is that if splitting to two PRs would be a trouble.

**Comment by ryelle** [comment]
@tyxla I am having a hard time pulling the rename out, since it happened in the middle of this work — I'd appreciate a review of this as-is, if you can. If it is too annoying, let me know and I'll try some more git magic.

File: client/me/purchases/purchases-site/index.jsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import CompactCard from 'components/card/compact';
+import { getSite } from 'state/sites/selectors';
+import PurchaseItem from '../purchase-item';
+import QuerySites from 'components/data/query-sites';
+import Site from 'blocks/site';
+import SitePlaceholder from 'blocks/site/placeholder';
+
+// Disconnected sites can't render the `Site` component, but there can be
+// purchases from disconnected sites. Here we spoof the Site header.
+const renderFauxSite = ( name, domain ) => {
```
Comment (tyxla) [inline client/me/purchases/purchases-site/index.jsx]: Maybe move that one to a separate stateless functional component?

**Comment by tyxla** [comment]
Not sure if this one's pending anything on the design side, but if it doesn't, feel free to 🚢

**Pull request merged**